vermouth/tests/test_molecule.py

```diff
@@ -345,3 +362,623 @@ def test_subgraph_edges(edges_between_molecule, edges_between_selections,
+    assert left != right
+
+
+@pytest.mark.parametrize('effector_class', (
+    vermouth.molecule.ParamDistance,
+    vermouth.molecule.ParamAngle,
+    vermouth.molecule.ParamDihedral,
+    vermouth.molecule.ParamDihedralPhase,
+))
+def test_link_parameter_effector_diff_keys(effector_class):
+    """
+    Test that LinkParameterEffector compare different if they have different keys.
+    """
+    n_keys = effector_class.n_keys_asked
+    left_keys = ['A{}'.format(idx) for idx in range(n_keys)]
+    right_keys = ['B{}'.format(idx) for idx in range(n_keys)]
+    left = effector_class(left_keys)
+    right = effector_class(right_keys)
+    assert left != right
+
+
+@pytest.mark.parametrize('left_class, right_class', itertools.combinations((
+    vermouth.molecule.ParamDistance,
+    vermouth.molecule.ParamAngle,
+    vermouth.molecule.ParamDihedral,
+    vermouth.molecule.ParamDihedralPhase,
+), 2))
+def test_link_parameter_effector_diff_class(left_class, right_class):
+    """
+    Test that LinkParameterEffector compare different if they have different classes.
+    """
+    left_n_keys = left_class.n_keys_asked
+    left_keys = ['A{}'.format(idx) for idx in range(left_n_keys)]
+    left = left_class(left_keys)
+
+    right_n_keys = right_class.n_keys_asked
+    right_keys = ['B{}'.format(idx) for idx in range(right_n_keys)]
+    right = right_class(right_keys)
+
+    assert left != right
+
+
+@st.composite
+def attribute_dict(draw, min_size=0, max_size=None):
+    keys = st.one_of(st.text(), st.integers(), st.none())
+    bases = (st.text(), st.integers(), st.floats(), st.none())
+    lists = st.lists(st.one_of(*bases, attribute_dict(max_size=1)), max_size=3)
+    values = st.one_of(*bases, lists, attribute_dict(max_size=1))
+    return draw(st.dictionaries(keys, values, min_size=min_size, max_size=max_size))
+
+
+@st.composite
+def random_interaction(draw, graph, natoms=None,
+                       interaction_class=Interaction, attrs=False):
+    if natoms is None:
+        natoms = draw(st.integers(min_value=0, max_value=4))
+    atoms = tuple(draw(st.sampled_from(list(graph.nodes))) for _ in range(natoms))
+    # TODO: Allow for LinkParameterEffector instances.
+    parameters = st.lists(elements=st.text())
+    meta = draw(st.one_of(st.none(), attribute_dict()))
+    if attrs:
+        atom_attrs = tuple(draw(st.fixed_dictionaries({})) for _ in atoms)
+        return interaction_class(
+            atoms=atoms,
+            atom_attrs=atom_attrs,
+            parameters=parameters,
+            meta=meta,
+        )
+    return interaction_class(atoms=atoms, parameters=parameters, meta=meta)
+
+
+@st.composite
+def interaction_collection(draw, graph,
+                           interaction_class=Interaction, attrs=False):
+    result = {}
+    ninteraction_types = draw(st.integers(min_value=0, max_value=1))
+    for _ in range(ninteraction_types):
+        ninteractions = draw(st.integers(min_value=0, max_value=1))
```

Comment:
min_value=1 to prevent no-ops

Reply:
We can have `molecule.interactions['something']` be an empty list. I'd like to have that case possible in the tests.

Reply:
Yes, but if `ninteractions=0`, `if type_name not in result and ninteractions > 0` below is False, and `type_name` doesn't get added to `result`

Reply:
It is a bug, not a feature :smile:

Reply:
Told you so ;)